src/olympia/addons/tasks.py

```diff
@@ -191,23 +193,50 @@ def recreate_theme_previews(addon_ids, **kw):
         '[%s@%s] Recreating previews for themes starting at id: %s...'
         % (len(addon_ids), recreate_theme_previews.rate_limit, addon_ids[0])
     )
-    addons = Addon.objects.filter(pk__in=addon_ids).no_transforms()
+    version_ids = Addon.objects.filter(pk__in=addon_ids).values_list('_current_version')
+    versions = Version.objects.filter(pk__in=version_ids)
```

Comment:
Probably want a `no_transforms()` in there still (`Version` does have its own default `transformer` that brings unnecessary stuff)

Reply:
if we're doing a full regeneration (`only_missing` is False, or there's one of more render sizes missing) then we do actually use the some of the stuff Version.transformer sets - `.all_files` so it depends.  I could add in `.no_transforms()` but it'd get more inefficient for the full regeneration use-case.

On the other hand, we don't even need the `Version` instance unless we're regenerating the preview (just resizing with `VersionPreview` can be filtered with `version_id`. (Changing _that_ to only load the Version inside the loop would make it even more inefficient for that case though.)

Reply:
Ok, let's leave it as-is